### ISSUE TITLE
Recognize more colors

### ISSUE THREAD

**Issue by rwols** [microsoft/vscode-json-languageservice#90]
Currently this server recognizes hex-encoded colors in the form of `#rrggbbaa`. Sublime text color schemes allow more variations. See https://www.sublimetext.com/docs/color_schemes.html#colors:ver-dev. These color schemes are written in a jsonc syntax. Ideally this server should parse those variations and present them via textDocument/documentColor.

It seems that vscode-css-language-server can already parse these kinds of variables. So that gives me an idea to port that code to vscode-json-languageservice, or perhaps they should share a common base.

Related: https://github.com/sublimelsp/LSP-json/issues/59

**Comment by KapitanOczywisty** [comment]
Good point. It's probably a matter of replacing [Provider](https://github.com/microsoft/vscode-json-languageservice/blob/07fec3faafa3f32915dcd6fcc6999a91ad113f88/src/services/jsonDocumentSymbols.ts#L239) and [Tests](https://github.com/microsoft/vscode-json-languageservice/blob/07fec3faafa3f32915dcd6fcc6999a91ad113f88/src/test/documentSymbols.test.ts#L272) with something similar to [CSS provider](https://github.com/microsoft/vscode-css-languageservice/blob/bfb6a187a3c38a30e5a6e8ce5145c03a06cf233a/src/services/cssNavigation.ts#L228) and [CSS tests](https://github.com/microsoft/vscode-css-languageservice/blob/bfb6a187a3c38a30e5a6e8ce5145c03a06cf233a/src/test/css/languageFacts.test.ts). It's probably not worth the effort to share this code, since there will be some differences, e.g. color names probably should work only if `format=color`.

**Comment by aeschli** [comment]
Right now, color tags appear when the scheme specifies  `format=colorHex`.
We would need a new format. maybe something like `format=colorCSS`.

I'm open to that, if you want to look into a PR...

**Comment by KapitanOczywisty** [comment]
In the [early drafts](https://tools.ietf.org/html/draft-zyp-json-schema-03#section-5.23) there was `format=color` proposed as css, but afik it never made into official schema. However I think it's a good name, and this is for color provider anyway, not for validation. Alternatively maybe we should check name for `color` or add key "color:true". I don't know how colors are written into the schema in the wild so not sure what is the best.
